sdk/go/common/util/rpcutil/writer.go

```diff
@@ -134,3 +117,33 @@ func MakeStreams(
 
 	return &nullCloser{}, stdout, stderr, nil
 }
+
+// Returns a pair of streams for use with the language runtimes InstallDependencies method
+func MakeInstallDependenciesStreams(
+	server pulumirpc.LanguageRuntime_InstallDependenciesServer,
+	isTerminal bool) (io.Closer, io.Writer, io.Writer, error) {
+
+	return makeStreams(
+		func(b []byte) error {
+			return server.Send(&pulumirpc.InstallDependenciesResponse{Stdout: b})
+		},
+		func(b []byte) error {
+			return server.Send(&pulumirpc.InstallDependenciesResponse{Stderr: b})
+		},
+		isTerminal)
+}
```

Comment:
Will `npm` continue to render a progress bar (re: #10901) on unix-y platforms, at least? It looks like this indirection shouldn't change the behavior.

It seems like it would be a pain to write a test to verify, so I'll just ask to confirm. (Not sure what we could test, could we snoop the stream and detect if it's sending VT commands?)

Reply:
Yeh this shouldn't change the current InstallDependencies behavior, I just ended up with a slightly different proto-layout. Given we update all these things in lock-step I could probably do a sanitize pass over this to bring everything into the same system, just need to do the dance with the yaml and java repos so wouldn't want to do it just before a release day.